src/app/components/SearchResultsPage/SearchResultsPage.jsx

```diff
@@ -81,23 +81,28 @@ class SearchResultsPage extends React.Component {
     return false;
   }
 
-  narrowSubjectFilters(apiFilters, selectedFilters) {
+  getSubjectLiteralFilters(apiFilters) {
     const apiSubjectLiteralFilters = apiFilters.filter(
       apiFilter => apiFilter.field === 'subjectLiteral',
     );
-    if (apiSubjectLiteralFilters.length) {
-      const subjectLiteralFilters = apiSubjectLiteralFilters[0];
+    return apiSubjectLiteralFilters.length ? apiSubjectLiteralFilters[0] : null;
+  }
+
+  subjectFilterIsSelected(selectedSubjectLiteralFilters) {
+    return subjectLiteralFilter =>
+      selectedSubjectLiteralFilters.some(
+        selectedFilter => subjectLiteralFilter.value === selectedFilter.value,
+      );
+  }
+
+  narrowSubjectFilters(apiFilters, selectedFilters) {
```

Comment:
just a small thing but on line 146, can you move the function after variable declarations?

Reply:
I don't understand this function, narrowSubjectFilters. Looks like you alter the values of subjectLiteralFilters but then I don't see you return the modified object. Where do you reference/use the modified object?

Reply:
The apiFilters are passed to the FilterPopUp component as a prop. The purpose of this function is to limit the subjectLiteral filters that get passed down to just those that have been selected. Otherwise FilterPopUp will display a long list of subjects in the filter dropdown. I don't really understand where these subjects come from.

Reply:
The long lists are the selectable filters I think. When you use filter (not subjectLiteral), you will see some selectable filters for a search result for you to refine your search. So the user is not only capable to remove filters but also add filters. Now, I am not sure subjectLiteral should apply the same concept.